packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2314,6 +2314,31 @@ export class ContainerRuntime
 		return result;
 	}
 
+	/**
+	 * Returns an aliased data store with the given alias.
+	 * @param alias - The alias for the data store.
+	 * @returns - The aliased data store as a FluidObject. Throws usage error if the data store
+	 * is not aliased or if it doesn't exist.
+	 */
+	public async getAliasDataStore(alias: string): Promise<FluidObject> {
+		await this.dataStores.waitIfPendingAlias(alias);
```

Comment:
I'd probably not wait for pending aliases here, this will make for more-symmetric behavior between the client setting the alias and others just searching for it.

Reply:
I mostly just did what `getRootDataStore` did. I will look into this. @andre4i do you have any thoughts on this?

Reply:
The point may be moot if we do the event thing. This is mainly for the local client to avoid creating new datastores and alias them if it already created one before, so it's just an optimization to avoid creating garbage datastores.

Reply:
i think we need to keep this. we had very bad behavior, 10000s of alias ops if we didn't check. aliasing is a race, but every client should only submit a single op in the race.

Reply:
Isn't this scenario that the local client is itself the one who has already submitted the alias op?  Why would they race themselves?

Reply:
@ChumpChief we had this exact problem happen before with one of our app partners, Tony is right (IIRC he was OCE when this happened) The client was attempting to create 1 singleton with 1000 calls (not judging the app behavior here, but this was a real-world situation). Each call checked to see if the datastore was created/aliased and if not, it would create the datastore and alias it, on the same client. What ended up happening was 1000 calls to create a datastore + 1000 calls to alias them. All datastore creation calls passed, while 999 calls for aliasing failed and one succeeded (as appropriate). Everything worked, however they ended up creating 999 useless datastores which would need to be GCed. By checking pending aliases, we can ensure that the local client will wait for the aliasing to complete instead of assuming there is no alias operation for that particular ID. This is the PR introducing that change: https://github.com/microsoft/FluidFramework/pull/10986.

Reply:
I don't feel overly strongly here, but I will say that customers will always be able to write bad patterns. If an individual client is racing itself 1000 times, then I'd argue they should probably have better internal coordination of their own singleton creation.

An alternate solution could be for us to provide an explicit method to check for a pending alias attempt.  Wouldn't be necessary if the customer keeps tight control over their singleton creation flows, but would be another option.

Reply:
We've instructed the customers how to work around this (basically a map of promises) but afterwards we made the decision to do it in the runtime layer ourselves as it's such an easy pattern to get wrong but with serious consequences.

Reply:
I disagree with you here @ChumpChief. In the encapsulated model we try to keep datastores isolated, and avoid the need for globals, so i don't think coordinating is that easy. Here we have a simple way to ensure the right behavior by keeping it to one race per alias per client, which has the advantage of keeping our api and the needs of the customer using our api simple. To me this a perfect example of the "pit of success" where customers get the right behavior by default, and maybe even without knowing it.

Reply:
I kept the `waitIfPendingAlias`.